### DISCUSSION TITLE
Reach limit of External Networks

### DISCUSSION THREAD

**Discussion by markhaines** [louislam/dockge#254] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

Not sure if this is a Dockge or a Docker limit (I wasn't experiencing it running my containers with regular docker compose though).  It looks like there is a limit of 30 x containers that have their own 'External Networks' automatically get created.  After which trying to start any additional containers generates an error:
`Error: failed to create network xyz_default: Error response from daemon: could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network`

### 👟 Reproduction steps

Create 30+ containers

### 👀 Expected behavior

Container #31+ should deploy as normal without issue

### 😓 Actual Behavior

Docker generates error as per description.

### Dockge Version

1.3.1

### 💻 Operating System and Arch

AlmaLinux 9.3

### 🌐 Browser

n/a

### 🐋 Docker Version

24.0.7

**Comment by louislam** [comment]
it should be a docker issue, dockge  is just running `docker compose up -d` for you. Nothing we can do.

**Reply by markhaines** [reply]
Is there a way I can change the config within Dockge to not generate the individual external networks (I don't need them).  Is it a case of defining that they all use the same default one or something?

**Reply by markhaines** [reply]
Ah, figured it out.  I need to define a default network and apply it to all the containers:

```
networks:
  default:
    name: my-default-network
    external: true
```

**Reply by thetredev** [reply]
You could also tell Docker to use more IP ranges, depending on your use case: https://stackoverflow.com/a/69027727
